distributed/tests/py3_test_client.py

```diff
@@ -37,3 +38,30 @@ def test_as_completed_async_for(c, s, a, b):
     yield f()
 
     assert set(results) == set(range(1, 11))
+
+
+def test_asyncio_await():
+    from tornado.platform.asyncio import to_asyncio_future
+    ioloop = asyncio.get_event_loop()
+
+    async def f():
+        scheduler = Scheduler(loop=ioloop)
+        scheduler.start('127.0.0.1')
+        worker = Worker(scheduler.address, loop=ioloop)
+        await to_asyncio_future(worker._start())
+
+        client = Client(scheduler.address, loop=ioloop, start=False)
+        await to_asyncio_future(client._start())
+
+        future = client.submit(inc, 1)
+        result = await future
+        assert result == 2
+
+        await to_asyncio_future(client._shutdown())
```

Comment:
I think it sounds like potentially a good idea.  I probably won't pursue this myself near-term.  Is this something that you would like to try out?